### ISSUE TITLE
Setting component props When Rendering Root Component

### ISSUE THREAD

**Issue by dschinkel** [rvdkooy/react-testutils-additions#21]
I'm getting an error when I try to render my root <App/> component because the sub component CompanyList underneath it expects me to be passing in a store value (data).  How do I get at sub components (such as my CompanyList component below) if I render the top component when using the react-testutils-additions?  

FYI, I need to render the top component because I'm also going to be testing that I hit certain react-router routes.  Right now the routes are defined in my <App/>.

So for example:

**myTests.js**

```
    it('should go to Company Detail Page when company is clicked', () => {

        //store.push([{id: 1, name: 'Some Company Name'}]); - I need to access a store property on my companyList react component.

        var foundComponent = renderIntoDocument(<App/>); // error (see below)
        Simulate.click(foundComponent[0]);
    });
```

**App.js**

```
import { Component } from 'reactivate';
import { IndexRoute, Router, Route, hashHistory } from 'react-router'
import CompanyList from './CompanyList';
import CompanyDetail from './CompanyDetail';

const Container = Component({
    render: function () {
        return (
            <div>{this.props.children}</div>
        );
    }
});

const App = Component({
    render() {
        return (
            <Router history={hashHistory}>
                <Route path="/" component={Container}>
                    <IndexRoute component={CompanyList}/>
                    <Route name="companyDetail" path="company/:companyId" component={CompanyDetail}/>
                </Route>
                <Route path="/" component={Container}></Route>
            </Router>
        );
    }
})

export default App;
```

**CompanyList.js**

```
import {Component,Store} from 'reactivate';
var companies = require('../../shared/data/companies.json');
var Link = require('react-router').Link;

const CompanyList = Component({
    store: Store('/companies'),
    handleClick: function onChange(event){
        this.setState({
            name: event.target.value
        });
    },
    render(){
        return (
            <div className="ft-companies">
            {
                this.store.value().map(company => {
                    return (
                        <p className="ft-company" key={company}>
                            <Link to="/company/1" params={{id: company.id}}>{company.name}</Link>
                        </p>
                    );
                })
            }
            </div>
        );
    }
});

export default CompanyList;
```

So in other words **I get the error** "`TypeError: Cannot read property 'map' of undefined`" because my CompanyList component expects me to set the store property `store: Store('/companies')` from my test.

**Comment by dschinkel** [comment]
---

**Comment by rvdkooy** [comment]
I'm not an expert on reactivate but shouldn't you implement a getInitialState in your CompanyList component?

It looks like Reactivate uses that to set the initialstate of your store
https://github.com/dbmeads/reactivate/blob/master/lib/Component.js#L7

**Comment by dschinkel** [comment]
No the way reactivate works is that it manages setting and updating its state internally.  You just pass state through the store prop.

**Comment by dschinkel** [comment]
Actually scratch my last comment

**Comment by rvdkooy** [comment]
is it fixed?

**Comment by dschinkel** [comment]
No I'll reply with a resolution soon.
